resources/views/filters/form.blade.php

```diff
@@ -23,13 +23,25 @@
             <div class="field flex flex-col gap-1 field-">
                 @if (is_array($field))
                     <label>{!! Arr::get($field, 'label', __('crud.fields.' . $field['field'])) !!}</label>
-                        <?php $model = null;
+                    <?php 
+                        $model = null;
                         $value = $filterService->single($field['field']);
                         if (!empty($value) && $field['type'] == 'select2') {
                             $modelclass = new $field['model'];
                             $model = $modelclass->find($value);
                         }
-                        ?>
+                        // Support for fields with multiple models selected
+                        if (Illuminate\Support\Arr::get($field, 'multiple') === true && $field['type'] == 'selectMultiple') {
+                            $value = $filterService->filterValue($field['field']);
+                            if (!empty($value)) {
+                                $modelclass = new $field['model'];
+                                $models = $modelclass->whereIn('id', $value)->get()->pluck('name', 'id')->toArray();
+                            } else {
```

Comment:
Instead of else unset, make it 4 lines shorter by declaring $models = null on line 27 `$model = $models = null`